jplephem/spk.py

```diff
@@ -48,7 +48,11 @@ def close(self):
         self.daf.file.close()
         for segment in self.segments:
             if hasattr(segment, '_data'):
-                del segment._data  # TODO: explicitly close each memory map
+                del segment._data
+        if self.daf._array is not None:
+            del self.daf._array
```

Comment:
Maybe I shouldn't attempt code review right before sleeping! Now that I'm awake again, I see immediately that instead of `del` I should have asked for the object to be restored to its original state, with `None` assigned to each of these attributes — it's much easier to program around objects where attributes always exist than to try using objects where attributes can appear and disappear. Thanks!

Reply:
And if we're just setting them to `None`, we don't even need the `is not None` check.